packages/flutter/lib/src/material/tab_controller.dart

```diff
@@ -243,6 +251,26 @@ class TabController extends ChangeNotifier {
   bool get indexIsChanging => _indexIsChangingCount != 0;
   int _indexIsChangingCount = 0;
 
+  /// The Duration used to animate [previousIndex] to [index] as a
+  /// consequence of calling [animateTo].
+  ///
+  /// This value is specifically set during the [animateTo] animation that's
+  /// triggered when the user taps a [TabBar] tab. It is intended to apply the
+  /// animation parameters to the [TabBarView]. It becomes null when [offset]
+  /// is changing as a result of the user dragging (and "flinging") the [TabBarView].
+  Duration? get indexChangingDuration => _indexChangingDuration;
```

Comment:
Can we avoid exposing transient state in this class?

Reply:
Imagine someone build code around these transient state, and then we have to make sure these transient behaviors are not to be modified without a breaking change.

Reply:
@chunhtai  How about make streamcontroller in tabarview and when tabcontroller.animateTo( params ) is executed, params is send to stream. AnimationTick is called in tabbarview and then before pagecontroller.animateToPage is executed, we use tabcontroller.animateTo() 's params from streamcontroller

Reply:
When the tabbarview _handleTabControllerAnimationTick is called, it actually can access the driving animation, not sure why it abstract away to only use index and guess the duration and curve by itself....

It seems to me we should make sure both PageController and TabController are driven by the same animation. We probably need to do some refactoring around this.

Reply:
@chunhtai You mean, just making a field in tabbar and then using a stream or something to share this field with pageController is a way to end up exposing transient state.

Besides refactoring animation logic is a appropriate way, right?

Sorry  for the late reply and Thank you for your advice and explanation. I learn a lot!